### ISSUE TITLE
Wrong checksum of Pkg.jl in release-1.0 branch

### ISSUE THREAD

**Issue by ronisbr** [JuliaLang/julia#30088]
Hi guys,

I am trying to build the openSUSE package of Julia 1.0.2. However, I am getting an error with the checksum of Pkg.jl. Take a look what I get:

```
$ wget https://api.github.com/repos/JuliaLang/Pkg.jl/tarball/d305e82fd353cb67e8a064800b9972ee1cb7b5e0                                                                      
$ md5sum d305e82fd353cb67e8a064800b9972ee1cb7b5e0                                                                                                                              
97bd2d6754259b90e7f96d7f71cfa84e  d305e82fd353cb67e8a064800b9972ee1cb7b5e0
```

However, the expected MD5 sum according to the file `julia/deps/checksum/Pkg-d305e82fd353cb67e8a064800b9972ee1cb7b5e0.tar.gz` is:

```
a633a375be6e5649865d8feee23bc177
```

**Comment by ronisbr** [comment]
In fact, something very weird is happening. The checksum of the downloaded file is changing every time. Now I get this:

```
md5 d305e82fd353cb67e8a064800b9972ee1cb7b5e0                    15:36:01
MD5 (d305e82fd353cb67e8a064800b9972ee1cb7b5e0) = 8b5e28693fa3655ff8268b767e731256
```

**Comment by StefanKarpinski** [comment]
I believe that GitHub generates tarballs on the fly in a non-deterministic way, so we probably should not be checksumming Pkg like this. Of course, we need to checksum it somehow but I don't think we can do so at the tarball stage, only later at the tree step which should match the requested tree.

**Comment by ronisbr** [comment]
The checksum is working for `libuv`, `libwhich`, and `utf8proc`, which are also being downloaded using Github API.

**Comment by ronisbr** [comment]
In the mean time, is there an easy way (i.e. without patching the Makefile) to disable the checksum verification so that I can build 1.0.2 to openSUSE?

**Comment by StefanKarpinski** [comment]
> The checksum is working for `libuv`, `libwhich`, and `utf8proc`, which are also being downloaded using Github API.

Yeah, I'm not sure what the difference is. FWIW, I'm consistently getting 8b5e28693fa3655ff8268b767e731256 as the MD5 sum for `Pkg`. But I only downloaded it a bunch of times in close succession so caching would explain that.

**Comment by vtjnash** [comment]
> The checksum is working for 

In that case, this may be fixed by 
https://github.com/JuliaLang/julia/pull/30075, as that alters the build to download Pkg in the same way as the rest of the deps

**Comment by ronisbr** [comment]
Guys, I am unable to build Julia at the moment. Does anyone know if this was fixed for the upcoming 1.0.3?

**Comment by fredrikekre** [comment]
Yes, should be fixed in 1.0.3.

**Comment by ronisbr** [comment]
This is fixed and can be closed! I could build successfully Julia 1.0.3 for openSUSE. Thanks!
